### PR TITLE
Restore Jetpack Earn Upsells and Settings for new Plans

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Change copy on `/earn` to use new Jetpack plan names
* replace upsell on `/earn/ads-earnings`

Untitled 4@0 5x

Screen Shot 2020-09-21 at 1 10 19 PM

#### Testing instructions

Test above routes with Free Jetpack plans and paid ( security daily will do ) 

1. Navigate to `/earn` with a WordPress.com Free, Jetpack Free, and new Jetpack Paid plan
2. Make sure the screens above match ( pat special attention to copy )
3. Go `/earn/ads-earnings` with a Jetpack Free Plan
4. Make sure the "upgrade" button takes you straight to the cart with security daily plan

### CODE REVIEW

File: client/my-sites/earn/ads/wrapper.jsx [hunk 1]
```diff
@@ -6,18 +6,18 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
+import { overSome } from 'lodash';
```
Comment (rcanepa) [inline client/my-sites/earn/ads/wrapper.jsx:9]: TIL. First time I see this `overSome` utility. Looks pretty handy.

**Comment by rcanepa** [comment]
What went well:
* `/earn` tested well with Jetpack Free and WordPress Free. 
* Upgrade button in `/earn/ads-earnings` with a Jetpack Free Plan.

What I'm not sure about (screenshot took with a site with Jetpack Security Daily (monthly):
* Collect PayPal payments CTA says "Unlock this feature" while at the same time the card says "Available with Jetpack Security and Jetpack Complete."
* Several cards display the "Available with Jetpack Security and Jetpack Complete." message and I don't know if that is correct or not while in some of them it appears to be hidden.
image

File: client/my-sites/earn/home.tsx [hunk 2]
```diff
@@ -100,14 +102,19 @@ const Home: FunctionComponent< ConnectedProps > = ( {
...
-			? ''
-			: ' ' + translate( 'Available only with a Premium or Professional plan.' );
+	const getAnyPlanNames = () => {
+		const jetpackText = translate( 'Available with Jetpack Security and Jetpack Complete.' );
 
 		// Space isn't included in the translatable string to prevent it being easily missed.
 		return isNonAtomicJetpack
-			? jetpackText
+			? ' ' + jetpackText
+			: ' ' + translate( 'Available with any paid plan.' );
```
Comment (a8ci18n) [inline client/my-sites/earn/home.tsx:111]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Available with any paid plan' )` *ES Score: 10*

File: client/my-sites/earn/ads/wrapper.jsx [hunk 3]
```diff
@@ -229,39 +235,29 @@ class AdsWrapper extends Component {
...
 		return (
 			<UpsellNudge
 				callToAction={ translate( 'Upgrade' ) }
-				plan={ PLAN_JETPACK_PREMIUM }
-				title={ translate( 'Upgrade to the Premium plan and start earning' ) }
+				plan={ PLAN_JETPACK_SECURITY_DAILY }
+				title={ translate( 'Upgrade and start earning' ) }
 				description={ translate(
-					"By upgrading to the Premium plan, you'll be able to monetize your site through the Jetpack Ads program."
+					'Make money each time someone visits your site by displaying ads on all your posts and pages.'
```
Comment (a8ci18n) [inline client/my-sites/earn/ads/wrapper.jsx:246]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Make money each time someone visits your site by displaying advertisements on all your posts and pages.' )` *ES Score: 8*

**Review by monsieur-z — APPROVED** [review]
It tests well!

Checking the feature (with `planHasFeature` or `hasFeature`) instead of the plan would probably make the addition of new plans easier in the future, but definitely not a blocker.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4512050

Thank you @cbauerman for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4512050).

**Pull request merged**